File: coffee_machine_python/coffee_machine.py
# coffee_machine.py
from ingredient_manager import IngredientManager
from beverage import Espresso, Latte

class CoffeeMachine:
    def __init__(self):
        self.ingredients = IngredientManager()

    def make_coffee(self, coffee_type):
        if coffee_type == 'espresso':
            coffee = Espresso()
        elif coffee_type == 'latte':
            coffee = Latte()
        else:
            print("Unknown coffee type.")
            return
        self.brew_coffee(coffee)

    def brew_coffee(self, coffee):
        if self.ingredients.has_ingredients(coffee):
            self.ingredients.use_ingredients(coffee)
            print(f"{coffee.name} is ready!")
        else:
            print(f"Not enough ingredients for {coffee.name}.")

File: coffee_machine_java/src/CoffeeMachine.java
// src/CoffeeMachine.java
public class CoffeeMachine {
    private IngredientManager ingredients;

    public CoffeeMachine() {
        ingredients = new IngredientManager();
    }

    public void makeCoffee(String coffeeType) {
        Beverage coffee;
        switch (coffeeType.toLowerCase()) {
            case "espresso":
                coffee = new Espresso();
                break;
            case "latte":
                coffee = new Latte();
                break;
            default:
                System.out.println("Unknown coffee type.");
                return;
        }
        brewCoffee(coffee);
    }

    private void brewCoffee(Beverage coffee) {
        if (ingredients.hasIngredients(coffee)) {
            ingredients.useIngredients(coffee);
            System.out.println(coffee.getName() + " is ready!");
        } else {
            System.out.println("Not enough ingredients for " + coffee.getName() + ".");
        }
    }

    // Getter method to access the IngredientManager
    public IngredientManager getIngredients() {
        return ingredients;
    }
}
